h2/src/main/org/h2/command/Parser.java

```diff
@@ -1572,6 +1572,20 @@ private Prepared parseShow() {
         } else if (readIf("DATESTYLE")) {
             // for PostgreSQL compatibility
             buff.append("'ISO' DATESTYLE");
+        } else if (readIf("SEARCH_PATH")) {
+            // for PostgreSQL compatibility
+            String[] searchPath = session.getSchemaSearchPath();
+            StringBuilder searchPathBuff = new StringBuilder();
+            if (searchPath != null) {
+                for (int i = 0; i < searchPath.length; i ++) {
+                    if (i > 0) {
+                        searchPathBuff.append(", ");
+                    }
+                    quoteIdentifier(searchPathBuff, searchPath[i], HasSQL.QUOTE_ONLY_WHEN_REQUIRED);
+                }
+                StringUtils.quoteStringSQL(buff, searchPathBuff.toString());
+            }
+            buff.append(" SEARCH_PATH");
```

Comment:
if `searchPath == null` an invalid SQL is generated (`SELECT SEARCH_PATH`) instead of expected `SELECT '' SEARCH_PATH`. Either append `''` explicitly (and move `searchPathBuff` into `if`), or move `StringUtils.quoteStringSQL(buff, searchPathBuff.toString());` outside of `if`.

Reply:
Sorry for the misplacement of `StringUtils.quoteStringSQL()` : - (